bin/lib/zopen-clone

```diff
@@ -0,0 +1,229 @@
+  fi
+else
+  printVerbose "Stripping any trailing '/' char"
+  clonefs="${clonefs%/}"
+  mkdir -p "$clonefs"
+fi
+
+#clonefs="$clonefs/"
+printInfo "- Cloning from '$rootfs' to '$clonefs'..."
+progressHandler "spinner" "- Clone complete" &
+ph=$!
+killph="kill -HUP $ph"
+addCleanupTrapCmd "$killph"
+#cp -fRT "$rootfs/"  Fails to keep some relative symlinks correct
+
+printDebug "Need to pax old dir and unpax to new dest to keep symlink structure"
+# Some files are set with permission -r--r--r-- which means they cannot be copied
+# over without some work.  Rather than search for these files, attempt to copy
+# and catch the errors, forcibly re-copying the file over afterwards. 
+# Temporary file location
+for tmp in "$TMPDIR" "$TMP" /tmp; do
+  [ ! -z "$tmp" ] && [ -d "$tmp" ] && break
+done
+[ ! -d "$tmp" ] && printError "Could not locate suitable temporary directory [tried \$TMPDIR \$TMP & /tmp]. Define a temporary location and retry command"
+FIFO_PIPE_STDOUT="$tmp/zopen_clone.stdout.pipe"
+[ ! -p "$FIFO_PIPE_STDOUT" ] || rm -f "$FIFO_PIPE_STDOUT"
+mkfifo "$FIFO_PIPE_STDOUT" && chtag -tc 819 "$FIFO_PIPE_STDOUT"
+cd "$rootfs" && pax -rw -p p "." "$clonefs"  2>>"$FIFO_PIPE_STDOUT" &
+while read OUTMSG; do
+set -x
+  printDebug "Parsing output: '$OUTMSG'"
+  #destFile=$(echo "$OUTMSG" | sed 's#.*FSUM7148.*"\([^"]+\)": EDC5111I.*#\1#')
+  destFile=$(echo "$OUTMSG" | sed 's#.*FSUM7148.*"\(.*\)".*EDC5111I.*#\1#')
+  if [ "$destFile" = "$OUTMSG" ]; then
+    printWarning "$destFile"
+  else
+    printDebug "Permission fail; trying to force copy to '$destFile'"
+    srcFile="$rootfs${destFile#"$clonefs"}"
+    printDebug "Copying sourcefile '$srcFile' to '$destFile'"
+    cp -f "$srcFile" "$destFile"
+    echo "$?"
+  fi
+  printError "Early exit"
+  #pax: FSUM7148 cannot create file "/u.../EventFacet/Render.pm": EDC5111I Permission denied.
+done < "$FIFO_PIPE_STDOUT"
+[ -n "$FIFO_PIPE_STDOUT" ] && [ -e "$FIFO_PIPE_STDOUT" ] && rm -f "$FIFO_PIPE_STDOUT"
+$killph 2>/dev/null  # if the timer is not running, the kill will fail
+
+printVerbose "Grabbing pkginstall location from within cloned env"
+if [ -e "$clonefs/etc/zopen/fstype" ]; then
+  zopen_pkginstall=$(cat "$clonefs/etc/zopen/fstype")
+else
+  printError "Unable to locate '$clonefs/etc/zopen/fstype'; unrecognisable file system. Check permissions"
+fi
+
+printDebug "Calculating certificate location"
+ZOPEN_CA_DIR="etc/pki/tls/certs"  # Mimic location on some Linux distributions
+certFileName="cacert.pem"
+ZOPEN_CA="$ZOPEN_CA_DIR/$certFileName"
+configFile="$clonefs/etc/zopen-config"
+printInfo "- Creating configuration for cloned environment"
+
+printDebug "Removing existing configuration file"
+[ -e "$configFile" ] && rm "$configFile"
+writeConfigFile "$configFile" "$clonefs" "$zopen_pkginstall" "$ZOPEN_CA"
+
+[ ! -r $configFile ] && printError "Config file '$configFile' was not created as expected. Correct any reported errors or run with --verbose option for more details"
+
+perms="744"
```

Comment:
Do we need to give ‘others’ read access?

Reply:
changed in latest commits